utils/material3/src/commonMain/kotlin/com/bumble/appyx/utils/material3/AppyxMaterial3NavNode.kt

```diff
@@ -0,0 +1,162 @@
+import androidx.compose.ui.zIndex
+import com.bumble.appyx.components.spotlight.Spotlight
+import com.bumble.appyx.components.spotlight.SpotlightModel
+import com.bumble.appyx.components.spotlight.SpotlightModel.State
+import com.bumble.appyx.components.spotlight.operation.activate
+import com.bumble.appyx.components.spotlight.ui.fader.SpotlightFader
+import com.bumble.appyx.interactions.core.plugin.Plugin
+import com.bumble.appyx.interactions.core.ui.Visualisation
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.navigation.Appyx
+import com.bumble.appyx.navigation.children.ChildAware
+import com.bumble.appyx.navigation.children.ChildAwareImpl
+import com.bumble.appyx.navigation.children.ChildEntry
+import com.bumble.appyx.navigation.composable.AppyxComponent
+import com.bumble.appyx.navigation.integration.LocalScreenSize
+import com.bumble.appyx.navigation.integration.ScreenSize.WindowSizeClass.COMPACT
+import com.bumble.appyx.navigation.modality.BuildContext
+import com.bumble.appyx.navigation.node.EmptyParentNodeView
+import com.bumble.appyx.navigation.node.Node
+import com.bumble.appyx.navigation.node.ParentNode
+import com.bumble.appyx.navigation.node.ParentNodeView
+
+@OptIn(ExperimentalMaterial3Api::class)
+open class AppyxMaterial3NavNode<NavTarget : Any>(
+    buildContext: BuildContext,
+    private val navTargets: List<NavTarget>,
+    private val navTargetResolver: (NavTarget) -> AppyxNavItem,
+    private val initialActiveElement: NavTarget,
+    private val animationSpec: SpringSpec<Float> = spring(
+        stiffness = Spring.StiffnessHigh
+    ),
+    private val visualisation: (UiContext) -> Visualisation<NavTarget, State<NavTarget>> = {
+        SpotlightFader(
+            uiContext = it,
+            defaultAnimationSpec = animationSpec
+        )
+    },
+    private val spotlight: Spotlight<NavTarget> = Spotlight(
+        model = SpotlightModel(
+            items = navTargets,
+            initialActiveIndex = navTargets.indexOf(initialActiveElement).toFloat(),
+            savedStateMap = buildContext.savedStateMap
+        ),
+        visualisation = visualisation
+    ),
+    view: ParentNodeView<NavTarget> = EmptyParentNodeView(),
+    childKeepMode: ChildEntry.KeepMode = Appyx.defaultChildKeepMode,
+    childAware: ChildAware<ParentNode<NavTarget>> = ChildAwareImpl(),
+    plugins: List<Plugin> = listOf(),
+) : ParentNode<NavTarget>(
+    appyxComponent = spotlight,
+    buildContext = buildContext,
+    view = view,
+    childKeepMode = childKeepMode,
+    childAware = childAware,
+    plugins = plugins
+) {
+
+    override fun resolve(navTarget: NavTarget, buildContext: BuildContext): Node =
+        navTargetResolver
+            .invoke(navTarget)
+            .node
+            .invoke(buildContext)
+
+    @Composable
+    override fun View(modifier: Modifier) {
+        val selectedIndex = spotlight.activeIndex.collectAsState(-1f).value.toInt()
```

Comment:
you'll be able to remove it if we use stateFlow
```suggestion
        val selectedIndex = spotlight.activeIxndex.collectAsState().value.toInt()
```